hs_file_types/models/base.py

```diff
@@ -595,3 +646,165 @@ def logical_delete(self, user, delete_res_files=True):
             # this should also delete on all metadata elements that have generic relations with
             # the metadata object
             metadata.delete()
+
+
+class BaseLogicalFile(models.Model):
+    """ base class for HydroShare file types """
+
+    logical_file_type = models.CharField(max_length=100, default="GenericLogicalFile")
+
+    # the dataset name will allow us to identify a logical file group on user interface
+    dataset_name = models.CharField(max_length=255, null=True, blank=True)
+
+    # this will be used for hsterms:dataType in resourcemetadata.xml
+    # each specific logical type needs to rest this field
+    data_type = "Generic data"
+
+    metadata = models.OneToOneField(BaseFileMetaData, related_name="logical_file")
+
+    @classmethod
+    def create(cls):
+        """this custom method MUST be used to create an instance of this class"""
+        generic_metadata = BaseFileMetaData.objects.create(file_metadata_type='Generic')
+        return cls.objects.create(metadata=generic_metadata)
+
+    @classmethod
+    def get_allowed_uploaded_file_types(cls):
+        # any file can be part of this logical file group - subclass needs to override this
+        return [".*"]
+
+    @classmethod
+    def get_allowed_storage_file_types(cls):
+        # can store any file types in this logical file group - subclass needs to override this
+        return [".*"]
+
+    @classmethod
+    def type_name(cls):
+        return cls.__name__
+
+    @property
+    def has_metadata(self):
+        return hasattr(self, 'metadata')
+
+    @property
+    def size(self):
+        # get total size (in bytes) of all files in this file type
+        return sum([f.size for f in self.files.all()])
+
+    @property
+    def resource(self):
+        res_file = self.files.all().first()
```

Comment:
This allows for polymorphic inconsistency of the detected resource files. Better to file resource here and then refer to it from each physical file.